tests/test_default_reply_headers.js

```diff
@@ -80,3 +134,65 @@ test('reply should not cause an error on header conflict', async t => {
   t.equal(headers['content-type'], 'application/xml')
   t.equal(body, '<html></html>')
 })
+
+test('default replay headers are not used when direct replay headers are a different case', async t => {
+  const scope = nock('http://example.com')
+    .defaultReplyHeaders({
+      'X-Powered-By': 'Meeee',
+      'X-Another-Header': 'Hey man!',
+    })
+    .get('/')
+    .reply(200, 'Success!', {
+      'X-Custom-Header': 'boo!',
```

Comment:
I used the two sets of two headers to show: a default header that is not overridden, a default header that is overridden, a reply header that is non-conflicting, and a reply header that overrides a default. 

I'm not really of fan of the random header field names. Maybe if I use names that actually describe why they're in the test it will make it clearer?

Reply:
In general given the complexity of this library and the extensive API surface, I find tests much easier to maintain and reason about when there is a clear intent about what is being tested, which is reflected in the title. I've spent a lot of time digging through old tests with titles like "reply works" that call various things and assert various other things, which is the worst. We can't really be too specific.

Do you feel having a single test that combines these various cases makes a stronger test? I'd prefer to either update the title to list all the behaviors being tested, or to split up the tests so it's easy to connect the title to the test code and the assertions.

Reply:
I do think that having this as single test makes a stronger assertion against the expected result. While there are other tests that show default headers show up if no reply headers are set and vise-versa, this test proves that when both are provided, and even overlap, that we don't lose the functionality test elsewhere. For example, if a default header is overridden we want to make sure not all default headers are lost, but we can't write a test for that without recreating this test.

So maybe a different name for the test? I'm horrible with naming, always open to suggestions. Or a comment at the top of the function outlining the cases?